### PR TITLE
Add CGI mode

### PR DESCRIPTION
**Pull request by rabbit-of-caerbannog**
Adds small changes to more easily support serving the script via CGI

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -33,7 +33,7 @@ Clone the git repository:
 	git clone https://github.com/RaymiiOrg/bash-http-monitoring.git
 	cd bash-http-monitoring
 
-Edit the `srvmon.sh` script and add your sites. A few examples are provided. This is the syntax:
```
Comment (RaymiiOrg) [inline README.md:36]: I'm not fond of removing the extension, could you please explain why that would be necessary for this request?

File: README.md [hunk 2]
```diff
@@ -54,11 +54,12 @@ Further global configuration options include:
 	maxConcurrentCurls=12 # How many curl checks to run at the same time
 	defaultTimeOut=10 # Max timeout of a check in seconds
 	flapRetry=5 # After how many seconds should we re-check any failed checks? (To prevent flapping)
-	title="Status Dashboard" # Title of the webpage
+	title="Status Dashboard" # Title of the webpage 
+	cgi=false # Enable or disable CGI mode
```
Comment (RaymiiOrg) [inline README.md]: This comment does not tell me what cgi mode does. Could you please add a small explanation comment what it is and why you would use it?

File: README.md [hunk 3]
```diff
@@ -68,10 +69,18 @@ If you want to set up a cronjob, send the output to a temp file and when finishe
...
 running. Like so:
 
-	* * * * * /bin/bash /opt/srvmon/srvmon.sh > /var/www/index.html.tmp && /bin/mv /var/www/index.html.tmp /var/www/index.html
+	* * * * * /bin/bash /opt/srvmon/srvmon > /var/www/index.html.tmp && /bin/mv /var/www/index.html.tmp /var/www/index.html
 
 If the check fails for whatever reason, the "old" page will not be overridden.
 
+### CGI mode
+
+If you want to set up CGI mode, you need to copy the script to your server CGI directory.
```
Comment (RaymiiOrg) [inline README.md:85]: Same here as the comment on the script cgi variable. I saw the reddit thread where this was discussed but a random user might not know what cgi is or why they would want that. Could you please add a few lines explaining what it is and the benefits or use cases?

File: README.md [hunk 4]
```diff
@@ -68,10 +69,18 @@ If you want to set up a cronjob, send the output to a temp file and when finishe
...
-	* * * * * /bin/bash /opt/srvmon/srvmon.sh > /var/www/index.html.tmp && /bin/mv /var/www/index.html.tmp /var/www/index.html
+	* * * * * /bin/bash /opt/srvmon/srvmon > /var/www/index.html.tmp && /bin/mv /var/www/index.html.tmp /var/www/index.html
 
 If the check fails for whatever reason, the "old" page will not be overridden.
 
+### CGI mode
+
+If you want to set up CGI mode, you need to copy the script to your server CGI directory. 
+You can use `docker` for this purpose. Like so:
+
```
Comment (RaymiiOrg) [inline README.md:87]: Could you also please add instructions for usage without docker?

File: srvmon [hunk 5]
```diff
@@ -220,6 +223,9 @@ cleanupOKCheckFiles() {
 
 
 writeHeader() {
+  if [[ "$cgi" = true ]]; then
```
Comment (RaymiiOrg) [inline srvmon]: Please use variable expansion with curly brackets to remain the same style as the rest of the script ( ${cgi} ). Also, does the single = work? (That would be new to me, I'd expect it to be $cgi == "true", but hey you keep learning every day)

**Review by RaymiiOrg — CHANGES_REQUESTED** [review]
Thank you for your pull request! I've added a few small comments, apart from that it looks good!

File: README.md (hunk 1 above)
Comment (rabbit-of-caerbannog) [reply README.md:36]: Mostly due to the configuration of the docker image I've picked:

https://github.com/hypoport/httpd-cgi/blob/master/httpd.conf#L71

File: README.md (hunk 2 above)
Comment (rabbit-of-caerbannog) [reply README.md]: Added below

File: README.md (hunk 3 above)
Comment (rabbit-of-caerbannog) [reply README.md:85]: Added

File: README.md (hunk 4 above)
Comment (rabbit-of-caerbannog) [reply README.md:87]: That's the above line. If you already have CGI enabled, you kinda just drop the script in a directory for it to work.

File: README.md (hunk 3 above)
Comment (RaymiiOrg) [reply README.md:85]: Thanks!

File: README.md (hunk 4 above)
Comment (RaymiiOrg) [reply README.md:87]: Okay thanks for the clarification. I've never used cgi scripts, a bit before my time

File: srvmon (hunk 5 above)
Comment (rabbit-of-caerbannog) [reply srvmon]: Done.

Like, all things, bash is weird. `[[` is a upgrade for `[`, which is almost an alias for `test`, which basically interprets the arguments on it's own (single `=` included). 

`==` is actually only supported on `[[`. It fails on both `[` and `test`, which simply use `=` for equality. This is also supported by `[[`.

File: README.md (hunk 1 above)
Comment (RaymiiOrg) [reply README.md:36]: Clear enough, thanks for clarifying

File: srvmon (hunk 5 above)
Comment (rabbit-of-caerbannog) [reply srvmon]: You can do these and variations on the command line to get a feeling

`test 1 = 2`
`[ 1 = 2 ]`
`[[ 1 = 2 ]]`

Comment (RaymiiOrg) [reply srvmon]: Thanks you for that explanation. I was unaware of that. Almost all of my scripts target recent versions of bash, not sh or other Unixes.

**Comment by RaymiiOrg** [comment]
Thank you for taking the time to respond to my comments! Merged 👍

**Pull request merged**